install/roles/create_vm/tasks/main.yml

```diff
@@ -0,0 +1,19 @@
+---
+- name: create a vm's
+  shell: . {{ overcloudrc }}; openstack server create --flavor pbench_flavor --image {{ guest_image_name }} --nic port-id={{ item[0].stdout }} --user-data {{ user_dir }}/pass_auth.sh {{ item[1] }}
+  with_together:
+    - "{{ port_ids.results }}"
+    - "{{ vm_names }}"
+  ignore_errors: yes
+
+- name: create a vm's
```

Comment:
Why are there two roles that are named the same and seem to do almost the same thing?

Reply:
There are two tasks for idempotency because
- In the first task  it creates Uperf client and server vm, so it uses UUID's of the ports  registered in var `port_ids` that were created newly if these ports didn't exist
- The second task creates uperf client and server vm's using the UUID of the ports  already exist registered in `get_port_ids` var

Reply:
Why not just have one task handle this?